trunk/src/modules/zoltan_wrapper.F90

```diff
@@ -134,11 +132,11 @@ subroutine zoltan_w_init()
       call zoltan_w_handle_err(ierr,'Zoltan_Set_Param')
 !
 !  ...set object weights to one floating point value per element
-      ierr = Zoltan_Set_Param(zz,'OBJ_WEIGHT_DIM','0')
+      ierr = Zoltan_Set_Param(zz,'OBJ_WEIGHT_DIM','1')
       call zoltan_w_handle_err(ierr,'Zoltan_Set_Param')
 !
 !  ...set edge weights to one floating point value per neighbor
-      ierr = Zoltan_Set_Param(zz,'EDGE_WEIGHT_DIM','0')
+      ierr = Zoltan_Set_Param(zz,'EDGE_WEIGHT_DIM','1')
```

Comment:
This change only makes sense if we add objects weights or edge weights other than `1.0`. I had commented that out in the routine because of a circular reference issue with `stc_get_nrdof`, did you add something like it back in?

Reply:
I have a separate module for my MG routines that does use edge weights. I could re-set it before I call it there (I set a re-initialize a bunch of parameters anyway). I don't think it hurts leaving it here but up to you.

Reply:
Did you code a function for returning weights that can be used here (`stc_nrdof` does that but I didn't want to copy-paste it somewhere just because of the circular reference issue)

Reply:
Not one that is easily adaptable to the non-MG case. A simple one might be to weight based on a power of the order, I guess that'd be better than nothing for hp meshes.

Reply:
Ok, I'll add that back in based on the `stc_get_nrdof` routine.